app/forms/nsm/steps/solicitor_declaration_form.rb

```diff
@@ -8,14 +8,27 @@ class SolicitorDeclarationForm < ::Steps::BaseFormObject
 
       def persist!
         Claim.transaction do
-          application.state = :submitted
-          application.update_work_item_positions!
-          application.update_disbursement_positions!
+          application.state = new_state
+          if application.state == 'submitted'
```

Comment:
This won't ever trigger, because the state will be `:submitted`, not `'submitted'`